### DISCUSSION TITLE
How to render an HTML comment using JSX?

### DISCUSSION THREAD

**Discussion by raed667** [solidjs/solid#745] [Q&A]
Is there a way to render an HTML comment through solid JSX? 

```html
<!-- Single line comment -->

<!--
Multi
line
comment
-->
```

I can do some hacking around with `renderToString` and `innerHTML` but it is very hacky and has a bunch of limitations.

**Comment by atk** [comment]
There are no comments in JSX: https://facebook.github.io/jsx/

However, in Solid, a component is supposed to return a DOM node on the client and a string for SSR. It shouldn't be too difficult to do that:

```js
const Comment = props =>
  isServer ? `\<!-- ${props.children} --\>` : globalThis.document?.createComment(children);
```

It might wreak havoc with `hydrate` though, as this uses comments to reference nodes.

**Reply by ryansolid** [reply]
I think there is chance of escaping from a parent component. I think you want:

```js
import { ssr } from "solid-js/web";

const Comment = props =>
  isServer ? ssr(`<!-- ${props.children} --\>`) : globalThis.document?.createComment(props.children);
```

**Reply by raed667** [reply]
Thanks both ! that looks clean !
